securedrop_export/export.py

```diff
@@ -0,0 +1,331 @@
+        if e:
+            try:
+                # If the file archive was extracted, delete before returning
+                if os.path.isdir(conf.tmpdir):
+                    print(conf.tmpdir)
+                    # shutil.rmtree(conf.tmpdir)
+                e_output = e.output
+            except Exception:
+                e_output = "<unknown exception>"
+            sys.stderr.write(e_output)
+            sys.stderr.write("\n")
+        # exit with 0 return code otherwise the os will attempt to open
+        # the file with another application
+        self.popup_message("Export error: {}".format(msg))
+        sys.exit(0)
+
+
+    def popup_message(self, msg):
+        try:
+            subprocess.check_call([
+                "notify-send",
+                "--expire-time", "3000",
+                "--icon", "/usr/share/securedrop/icons/sd-logo.png",
+                "SecureDrop: {}".format(msg)
+            ])
+        except subprocess.CalledProcessError as e:
+            msg = "Error sending notification:"
+            self.exit_gracefully(msg, conf, e=e)
+
+
+    def extract_tarball(self):
+        try:
+            with tarfile.open(self.archive) as tar:
+                tar.extractall(self.archive_path)
+        except Exception as e:
+            print (e)
+            msg = "Error opening export bundle: "
+            self.exit_gracefully(msg, e=e)
+
+
+    def unlock_luks_volume(self, encryption_key):
+        # the luks device is not already unlocked
+        if not os.path.exists(os.path.join("/dev/mapper/", self.encrypted_device)):
+            p = subprocess.Popen(
+                ["sudo", "cryptsetup", "luksOpen", self.device, self.encrypted_device],
+                stdin=subprocess.PIPE,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+            )
+            p.communicate(input=str.encode(encryption_key, "utf-8"))
+            rc = p.returncode
+            if rc != 0:
+                msg = "Bad passphrase or luks error."
+                self.exit_gracefully(msg)
+
+
+    def mount_volume(self):
+        # mount target not created
+        if not os.path.exists(self.mountpoint):
+            subprocess.check_call(["sudo", "mkdir", self.mountpoint])
+        try:
+            subprocess.check_call(
+                [
+                    "sudo",
+                    "mount",
+                    os.path.join("/dev/mapper/", self.encrypted_device),
+                    self.mountpoint,
+                ]
+            )
+            subprocess.check_call(
+                [
+                    "sudo", 
+                    "chown", 
+                    "-R", "user:user", self.mountpoint,
+                ]
+            )
+        except subprocess.CalledProcessError as e:
+            # clean up
+            subprocess.check_call(["sudo", "cryptsetup", "luksClose", self.encrypted_device])
+            subprocess.check_call(["rm", "-rf", self.tmpdir]) # TODO - why not srm?
```

Comment:
may be redundant given https://github.com/freedomofpress/securedrop-export/pull/1/files#diff-e41a261ab839c72ca53ae32069e0670fR18 ?

Reply:
Yeah, the exit_gracefully will cover this, and if not, the entrypoint function has it.